budget/utils.py

```diff
@@ -86,25 +88,33 @@ def list_of_dicts2json(dict_to_convert):
     """Take a list of dictionnaries and turns it into
     a json in-memory file
     """
-    bytes_io = BytesIO()
-    bytes_io.write(dumps(dict_to_convert))
-    bytes_io.seek(0)
-    return bytes_io
+    return BytesIO(dumps(dict_to_convert).encode('utf-8'))
 
 def list_of_dicts2csv(dict_to_convert):
     """Take a list of dictionnaries and turns it into
     a csv in-memory file, assume all dict have the same keys
     """
-    bytes_io = BytesIO()
+    # CSV writer has a different behavior in PY2 and PY3
```

Comment:
The thing is that in py3, even if you pass a file opened in binary mode to csv.writer and then try to write bytes with csv.writerow, you get a TypeError. 
```
Python 3.5.2 (default, Sep 14 2016, 11:28:32)
>>> import csv
>>> from io import BytesIO
>>> csv_file = BytesIO()
>>> writer = csv.writer(csv_file)
>>> writer.writerow([b"test", b"test2"])
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: a bytes-like object is required, not 'str'
```
Six.BytesIO is not a good candidate neither because it is equivalent to StringIO in py2 and BytesIO in py3 and we need the opposite. So for me, there is no other choice than using StringIO in py3 and BytesIO in py2.